lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -721,15 +715,18 @@ def on_tls_conf_set(
                 truststore_pwd=truststore_pwd,
                 keystore_pwd=keystore_pwd,
             )
-        else:
+
             # write the admin cert conf on all units, in case there is a leader loss + cert renewal
-            self.opensearch_config.set_admin_tls_conf(current_secrets)
+            if not admin_secrets.get("subject"):
```

Comment:
Why specific query on the `subject`?  Why not on the cert ifself or so?

Reply:
(Whatever the answer may be, perhaps it should go in the docstring, if the check is correct and fully intentional :-) )

Reply:
I'm a bit wondering on a verbose-named little function that should rather go in the `opensearch_tls` module instead... Like `has_valid_admin_cert()` or so...? Instead of the raw check

Reply:
The subject is checked because it's what is required to be put into the config in this case. This check makes sure that the `set_admin_tls_conf` in `opensearch_tls` has valid data to put into the config.

You're right, in general it would make sense to have a function that checks it and is then called multiple times throughout the code. Problem here is, that is changing what can be considered valid and it's also changing what is in the secret, both depending on the workflow. There might be the csr fist, then the cert and the ca and chain, later it also requires the subject and at a certain point also the keystore/truststore password. But all of these are different stages, so it will be pretty hard to make this valid at all times.